services/filemanager/filemanager.go

```diff
@@ -87,8 +87,27 @@ func (factory *FileManagerFactoryT) New(settings *SettingsT) (FileManager, error
 	return nil, fmt.Errorf("%w: %s", rterror.InvalidServiceProvider, settings.Provider)
 }
 
-// GetProviderConfigForBackupsFromEnv returns the provider config
-func GetProviderConfigForBackupsFromEnv(ctx context.Context) map[string]interface{} {
+// GetBackupStorageConfig returns the backup storage config based on the user settings
+func GetBackupStorageConfig(ctx context.Context) (string, map[string]interface{}) {
+	backendconfig.DefaultBackendConfig.WaitForConfig(ctx)
+	// TODO: Adapt for multi-tenancy
+	workspaceID := backendconfig.DefaultBackendConfig.GetWorkspaceIDForWriteKey("")
+	useRudderStorage := backendconfig.DefaultBackendConfig.GetDataRetentionSettingsForWorkspaceID(workspaceID).UseRudderServerStorage
```

Comment:
What if the control-plane runs on an older version, were this configuration is not defined?

Also, what if a bug is introduced in control-plane and this configuration is missing? We will interpret it as false, is this a safe option here?

Otherwise, we might want to modify 
```
GetDataRetentionSettingsForWorkspaceID(workspaceID)
```
to return an error if the setting is missing. To detect if Setting is missing we need to use a pointer when parsing the JSON.

Reply:
[Here](https://github.com/rudderlabs/rudder-server/pull/2453#discussion_r973991488) I proposed `(DataRetention, bool)` instead of the pointer as it seemed easy enough to copy across the stack instead of spreading pointers. The point is the same anyway, I'm cool with both options.

Reply:
[This idea](https://github.com/rudderlabs/rudder-server/pull/2453#discussion_r974265156) seems better to me instead of checking for nested nils. We can work with control plane team to invert the names. What do you guys think?

Reply:
It's not a bad idea and I'm OK with that, but personally I find it less robust as it might not be immediate for devs to understand the behaviour. On the other hand `(DataRetention, bool)` is unequivocally clear. It gives you the freedom to rename things as you want and don't depend on the control plane. Also you don't rely on zero values that sometimes can be missed or not taken into consideration. If you guys prefer zero-values or pointers it is also cool of course, just expressing a preference.